populate_test.go

```diff
@@ -158,6 +158,42 @@ func TestPopulate(t *testing.T) {
 		assert.False(t, targets.V1 == targets.V2, "fields should be different")
 	})
 
+	t.Run("annotated populate", func(t *testing.T) {
+		t.Parallel()
+
+		type result struct {
+			Out
+
+			V1 *t1 `name:"n1"`
```

Comment:
can you add a test case where the output is provided via fx.Annotate instead of an Out struct?